From 8038d0fcf0bebfb208e2cd3435d936e5bfd667a8 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Tue, 13 Dec 2016 18:55:40 +0200
Subject: [PATCH 1169/2241] Revert "fix: arm64: exception: ignore EL0
 asynchronous external abort"

This patch reverts commit 337b578a27da ("fix: arm64: exception: ignore EL0
asynchronous external abort").

PCIe0 CP1 window was not configured in ATF, causing EL0 asynchronous
external abort in Linux.
The PCIe issue was fixed in ATF (the windows are configured appropriately
now) so this work-around can be removed.

Change-Id: I809f842ebb76a2e1917946d759c23b56db18e280
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34656
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/kernel/entry.S | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/arch/arm64/kernel/entry.S b/arch/arm64/kernel/entry.S
index db206c3..698c4c6 100644
--- a/arch/arm64/kernel/entry.S
+++ b/arch/arm64/kernel/entry.S
@@ -227,14 +227,7 @@ ENTRY(vectors)
 	ventry	el0_sync			// Synchronous 64-bit EL0
 	ventry	el0_irq				// IRQ 64-bit EL0
 	ventry	el0_fiq_invalid			// FIQ 64-bit EL0
-#ifdef CONFIG_IGNORE_ASYNC_ABORT
-	/* If CONFIG_IGNORE_ASYNC_ABORT is enabled, in a case of exeption,
-	 * ignore it be performing eret (in el0_ignore).
-	 */
-	ventry	el0_ignore
-#else
 	ventry	el0_error_invalid		// Error 64-bit EL0
-#endif
 
 #ifdef CONFIG_COMPAT
 	ventry	el0_sync_compat			// Synchronous 32-bit EL0
@@ -306,10 +299,6 @@ ENDPROC(el1_error_invalid)
 el1_ignore:
 	eret
 ENDPROC(el1_ignore)
-
-el0_ignore:
-	eret
-ENDPROC(el0_ignore)
 #endif
 
 /*
-- 
2.7.4

